bottlerocket/samples/Makefile.toml

```diff
@@ -0,0 +1,248 @@
+ECS_WORKLOAD_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/ecs-workload-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["ECS_WORKLOAD_AGENT_IMAGE_URI"] } }
+K8S_WORKLOAD_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/k8s-workload-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["K8S_WORKLOAD_AGENT_IMAGE_URI"] } }
+SONOBUOY_TEST_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/sonobuoy-test-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["SONOBUOY_TEST_AGENT_IMAGE_URI"] } }
+EKS_RESOURCE_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/eks-resource-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["EKS_RESOURCE_AGENT_IMAGE_URI"] } }
+VSPHERE_K8S_CLUSTER_RESOURCE_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/vsphere-k8s-cluster-resource-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["VSPHERE_K8S_CLUSTER_RESOURCE_AGENT_IMAGE_URI"] } }
+VSPHERE_VM_RESOURCE_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/vsphere-vm-resource-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["VSPHERE_VM_RESOURCE_AGENT_IMAGE_URI"] } }
+CONTROLLER_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/controller:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["CONTROLLER_IMAGE_URI"] } }
+
+[tasks.metadata-url.env]
+PSEUDO_METADATA_URL = {script=["echo ${METADATA_BASE_URL}/${VARIANT}/${ARCHITECTURE}"]}
+
+[tasks.cluster-name.env]
+PSEUDO_CLUSTER_NAME = {script=["echo $(echo ${ARCHITECTURE} | tr '_' '-')-$(echo ${VARIANT} | tr -d '.')"]}
+
+[tasks.ova-name.env]
+PSEUDO_OVA_NAME = {script=["echo bottlerocket-${VARIANT}-x86_64-${STARTING_VERSION}.ova"]}
+
+[tasks.bottlerocket-ami-id.env]
+PSEUDO_BOTTLEROCKET_AMI_ID = {script=['''
+if [ $(echo ${VARIANT} | tr "-" " " | awk '{print $1}') = "aws" ]; then
+    if [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "migration" ]; then
+        AMI_VERSION=$(echo ${STARTING_VERSION} | tr -d "v")
+    else
+        AMI_VERSION="latest"
+    fi
+    aws ssm get-parameter \
+        --region ${AWS_REGION} \
+        --name "/aws/service/bottlerocket/${VARIANT}/$(echo ${ARCHITECTURE} | sed -e 's/aarch64/arm64/g')/${AMI_VERSION}/image_id" \
+        --query Parameter.Value --output text 
+fi
+''']}
+
+[tasks.gpu.env]
+PSEUDO_GPU = {script=['''
+if [ $(echo ${VARIANT} | tr "-" " " | awk '{print $NF}') = "nvidia" ]; then
+    echo true
+else
+    echo false
+fi
+''']}
+
+[tasks.mgmt-cluster-kubeconfig-base64.env]
+MGMT_CLUSTER_KUBECONFIG_BASE64 = {script=['''
+if [ -n "${MGMT_CLUSTER_KUBECONFIG_PATH}" ]; then
+    cat ${MGMT_CLUSTER_KUBECONFIG_PATH} | base64
+fi
+''']}
+
+[tasks.set-pseudo-env]
+dependencies = ["set-agent-images", "cluster-name", "gpu", "metadata-url", "ova-name", "bottlerocket-ami-id"]
+
+[tasks.set-env]
+dependencies = ["set-pseudo-env", "mgmt-cluster-kubeconfig-base64"]
+
+[tasks.set-env.env]
+CLUSTER_NAME = { value = "${PSEUDO_CLUSTER_NAME}", condition = { env_not_set = ["CLUSTER_NAME"] } }
```

Comment:
We're assigning the value in `PSEUDO_CLUSTER_NAME` to `CLUSTER_NAME`, if `CLUSTER_NAME` is not set in advance by the user. It works correctly when I test exporting `CLUSTER_NAME` before running the make command. Is my reasoning incorrect?

Reply:
I don't see why you can't just assign to `CLUSTER_NAME` instead of `PSEUDO_CLUSTER_NAME`.

Reply:
I couldn't use the `{ value = ..., condition = ...}` syntax because the value is being generated by some commands. Since I couldn't use that syntax, I had to unconditionally assign it to an environment variable, which would've overwritten the user-defined `CLUSTER_NAME`. That's why I assign it to a pseudo variable first, which is then conditionally passed into `CLUSTER_NAME` (only if the user didn't define that first).

Reply:
^ Ignore that, added conditional statements so that the PSEUDO variables could be removed.